File: C++/check-if-array-is-sorted-and-rotated.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    bool check(vector<int>& nums) {
        for (int i = 0, count = 0; i < size(nums); ++i) {
            if (nums[i] > nums[(i + 1) % size(nums)] && ++count > 1) {
                return false;
            }
        }
        return true;
    }
};

File: Python/check-if-array-is-sorted-and-rotated.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def check(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        count = 0
        for i in xrange(len(nums)):
            if nums[i] > nums[(i+1)%len(nums)]:
                count += 1
                if count > 1:
                    return False
        return True
